--- a/libpoolprotocols/testing/SimulateClient.cpp
+++ b/libpoolprotocols/testing/SimulateClient.cpp
@@ -87,6 +87,7 @@ void SimulateClient::workLoop()
     current.header = h256::random();
     current.block = m_block;
     current.boundary = h256(dev::getTargetFromDiff(m_difficulty));
+    current.epoch = current.block / 2147483647;  // EPOCH_LENGTH = 2^31 - 1
     m_onWorkReceived(current);  // submit new fake job

     cnote << "Using block " << m_block << ", difficulty " << m_difficulty;
